src/scheduler/MpiWorld.cpp

```diff
@@ -425,11 +452,14 @@ void MpiWorld::send(int sendRank,
             getLocalQueue(sendRank, recvRank)->enqueue(std::move(m));
         }
     } else {
-        logger->trace("MPI - send remote {} -> {}", sendRank, recvRank);
+        logger->debug("MPI - send remote {} -> {}", sendRank, recvRank);
 
-        // TODO - avoid creating a client each time?
-        scheduler::FunctionCallClient client(otherHost);
-        client.sendMPIMessage(m);
+        if (messageType == faabric::MPIMessage::RMA_WRITE) {
```

Comment:
RMA exposes a purely synchronous API, whereas for other MPI calls, only active things like `MPI_Recv` will block. With the new asyncrhonous client-server, the way to replicate purely synchronous behaviour is by managing the lifecycle of the client: it's destructor acts as synchronisation point.

Reply:
Could we somehow abstract this by having a function called `sendSynchronousMPIMessage` that contains these two lines and a comment explaining why? That way we can change it in future.

I think perhaps we would create a separate RPC just for handling MPI RMA.